designer/server/src/main/scala/pl/touk/nussknacker/ui/api/app/AppApiEndpoints.scala

```diff
@@ -0,0 +1,211 @@
+      .get
+      .in("app" / "healthCheck" / "process" / "validation")
+      .out(statusCode(Ok))
+      .out(jsonBody[HealthCheckProcessSuccessResponseDto.type])
+      .errorOut(statusCode(InternalServerError))
+      .errorOut(jsonBody[HealthCheckProcessErrorResponseDto])
+
+  def buildInfoEndpoint: Endpoint[Unit, Unit, Unit, BuildInfoDto, Any] =
+    baseNuApiPublicEndpoint
+      .get
+      .in("app" / "buildInfo")
+      .out(statusCode(Ok))
+      .out(jsonBody[BuildInfoDto])
+
+  def serverConfigEndpoint(implicit user: LoggedUser): Endpoint[LoggedUser, Unit, ServerConfigInfoErrorDto, ServerConfigInfoDto, Any] =
+    baseNuApiUserSecuredEndpoint(user)
+      .get
+      .in("app" / "config")
+      .out(statusCode(Ok))
+      .out(jsonBody[ServerConfigInfoDto])
+      .errorOut(statusCode(Forbidden))
+      .errorOut(plainBody[ServerConfigInfoErrorDto])
+
+  def userCategoriesWithProcessingTypesEndpoint(implicit user: LoggedUser): Endpoint[LoggedUser, Unit, Unit, UserCategoriesWithProcessingTypesDto, Any] =
+    baseNuApiUserSecuredEndpoint(user)
+      .get
+      .in("app" / "config" / "categoriesWithProcessingType")
+      .out(statusCode(Ok))
+      .out(jsonBody[UserCategoriesWithProcessingTypesDto])
+
+  def processingTypeDataReloadEndpoint(implicit user: LoggedUser): Endpoint[LoggedUser, Unit, ProcessingTypeDataReloadErrorDto, Unit, Any] =
+    baseNuApiUserSecuredEndpoint(user)
+      .post
+      .in("app" / "processingtype" / "reload")
+      .out(statusCode(NoContent))
+      .errorOut(statusCode(Forbidden))
+      .errorOut(plainBody[ProcessingTypeDataReloadErrorDto])
+
+  object Dtos {
+
+    object HealthCheckProcessSuccessResponseDto
+
+    final case class HealthCheckProcessErrorResponseDto(message: Option[String],
+                                                        processes: Option[Set[String]])
+
+    final case class BuildInfoDto(name: String,
+                                  gitCommit: String,
+                                  buildTime: String,
+                                  version: String,
+                                  processingType: Map[String, Map[String, String]],
+                                  otherProperties: Map[String, String])
+
+    final case class ServerConfigInfoDto(configJson: Json)
+    sealed trait ServerConfigInfoErrorDto
+    object ServerConfigInfoErrorDto {
+      case object AuthorizationServerConfigInfoErrorDto extends ServerConfigInfoErrorDto
+    }
+
+    final case class UserCategoriesWithProcessingTypesDto(map: Map[String, String])
+
+    sealed trait ProcessingTypeDataReloadErrorDto
+    object ProcessingTypeDataReloadErrorDto {
+      case object AuthorizationProcessingTypeDataReloadErrorDto extends ProcessingTypeDataReloadErrorDto
+    }
+
+    private[AppApiEndpoints] object Codecs {
+
+      implicit val healthCheckProcessSuccessResponseDtoCodec: io.circe.Codec[HealthCheckProcessSuccessResponseDto.type] = {
+        io.circe.Codec.from(
```

Comment:
Why did you use a qualified class name?

Reply:
because of the clash with Tapir's Codec. It's been renamed to CirceCodec now.